src/apispec/ext/marshmallow/openapi.py

```diff
@@ -163,7 +163,23 @@ def field2type_and_format(self, field):
         :param Field field: A marshmallow field.
         :rtype: dict
         """
-        type_, fmt = self.field_mapping.get(type(field), ("string", None))
+
+        #
+        # If this type isn't directly in the field mapping then check the
+        # heirachy until we find something that does.
+        #
+        for field_class in type.mro(type(field)):
```

Comment:
Not exactly sure of the difference, but perhaps `inspect.getmro` is preferred?

Reply:
Why not

```py
    type(field).__mro__
```

Or even 

```py
    field.__class__.__mro__
```

Is it discouraged to access those attributes?

We do it already:

https://github.com/marshmallow-code/apispec/blob/0845c66b67998f8e47b624c10efebe7bd5dbd532/src/apispec/ext/marshmallow/common.py#L120

Reply:
OK, apparently, using `type` is preferred over accessing `__class__` (https://stackoverflow.com/questions/9610993/python-type-or-class-or-is). Unless we're using old style PY2 classes, which shouldn't be the case if inheriting `Field`.

I didn't see any advice against accessing `__mro__` (https://stackoverflow.com/questions/2010692/what-does-mro-do), and I think it is more efficient than calling `mro()` (which does not matter in the context).

Reply:
Thanks for doing the research @lafrech. So it sounds like ` type(field).__mro__` is preferred. Let's go with it.

Reply:
Yeah, no big deal, all of these are functional anyway.

I can change `schema.__class__` into `type(schema)` someday.